azure_pipelines/templates/tests-with-conf-file.yml

```diff
@@ -58,12 +59,22 @@ steps:
       targetType: 'inline'
       script: |
         ls build/Build/Products/
-        xcodebuild test-without-building \
-            -xctestrun '${{ parameters.full_path }}' \
-            -destination '${{ parameters.destination }}' \
-            -retry-tests-on-failure \
-            -parallel-testing-enabled NO \
-            -resultBundlePath '$(Agent.BuildDirectory)/s/test_output/report.xcresult'      
+        if ['${{ parameters.retry_tests }}' == true]; then
```

Comment:
what is this change for?

Reply:
The `xcodebuild` flag "-retry-tests-on-failure " doesn't work properly in conjunction with the "Maximum Test Repetitions" from the xctestplan, because it marks the suite as a failure as opposed an individual test, which is then retried and passed. I had to do this for the E2E tests to have them pass if they get retried and pass. For the UI tests I've left the implementation as before.
The flag switches between one script or the other.

Reply:
Does this impact MSAL Objective-C in some ways?
Can you please add a comment explaining this?

Reply:
This doesn't impact because the old command is still on the "retry-tests" true path and by default I've set it to true.
Added comment to explain the IF